sources/src/resources/init-scripts/gradle-actions.inject-develocity.init.gradle

```
@@ -107,21 +108,35 @@ if (GradleVersion.current() < GradleVersion.version('6.0')) {
                     logger.quiet("Applying $BUILD_SCAN_PLUGIN_CLASS via init script")
                     applyPluginExternally(pluginManager, BUILD_SCAN_PLUGIN_CLASS)
                     if (geUrl) {
-                        logger.quiet("Connection to Develocity: $geUrl, allowUntrustedServer: $geAllowUntrustedServer")
+                        logger.quiet("Connection to Develocity: $geUrl, allowUntrustedServer: $geAllowUntrustedServer, captureTaskInputFiles: $geCaptureTaskInputFiles")
                         buildScan.server = geUrl
                         buildScan.allowUntrustedServer = geAllowUntrustedServer
                     }
                     buildScan.publishAlways()
+                    if (isAtLeast(gePluginVersion, '2.1') && GradleVersion.current() > GradleVersion.version('5.0')) {
+                        if (isNotAtLeast(gePluginVersion, '3.7')) {
+                            buildScan.captureTaskInputFiles = geCaptureTaskInputFiles
+                        } else {
+                            buildScan.capture.taskInputFiles = geCaptureTaskInputFiles
+                        }
+                    }
```

I think this would be a minor readability improvement:

```suggestion
                    if (GradleVersion.current() > GradleVersion.version('5.0')) {
                        if (isAtLeast(gePluginVersion, '3.7')) {
                            buildScan.capture.taskInputFiles = geCaptureTaskInputFiles
                        } else if (isAtLeast(gePluginVersion, '2.1')) {
                            buildScan.captureTaskInputFiles = geCaptureTaskInputFiles
                        }
                    }
```

To me, that makes it immediately clear which path to take based on plugin version.

---

There are several other places in the code where it could be applied.

---

Agree, will change